rust/lance-encoding/src/encodings/physical/basic.rs

```diff
@@ -193,7 +194,15 @@ impl ArrayEncoder for BasicEncoder {
     fn encode(&self, arrays: &[ArrayRef], buffer_index: &mut u32) -> Result<EncodedArray> {
         let (null_count, row_count) = arrays
             .iter()
-            .map(|arr| (arr.null_count() as u32, arr.len() as u32))
+            .map(|arr| {
+                if matches!(arr.data_type(), DataType::Null) {
+                    // Arrays with the null datatype report 0 as the null count so we
+                    // need special logic to get the correct null count
+                    (arr.len() as u32, arr.len() as u32)
+                } else {
+                    (arr.null_count() as u32, arr.len() as u32)
+                }
```

Comment:
If you are interested, I think all the exceptions on null count are described here: https://docs.rs/arrow/latest/arrow/array/trait.Array.html#method.logical_nulls